osu.Game/Rulesets/Objects/Pooling/PoolableDrawableWithLifetime.cs

```diff
@@ -26,17 +27,18 @@ public abstract class PoolableDrawableWithLifetime<TEntry> : PoolableDrawable wh
         /// </summary>
         protected bool HasEntryApplied { get; private set; }
 
-        // Drawable's lifetime gets out of sync with entry's lifetime if entry's lifetime is modified.
-        // We cannot delegate getter to `Entry.LifetimeStart` because it is incompatible with `LifetimeManagementContainer` due to how lifetime change is detected.
         public override double LifetimeStart
         {
             get => base.LifetimeStart;
             set
             {
-                base.LifetimeStart = value;
+                if (Entry == null && LifetimeStart != value)
+                    throw new InvalidOperationException($"Cannot modify lifetime of {nameof(PoolableDrawableWithLifetime<TEntry>)} when entry is not set");
 
-                if (Entry != null)
-                    Entry.LifetimeStart = value;
+                if (Entry == null) return;
```

Comment:
The way this is written feels strange/backwards to me. Wouldn't it be better reading written as

```csharp
if (LifetimeEnd == value)
    return;

if (Entry == null)
    throw new InvalidOperationException($"Cannot modify lifetime of {nameof(PoolableDrawableWithLifetime<TEntry>)} when entry is not set");
```

Reply:
That changes the behavior because setting lifetime while `HitObjectLifetimeEntry.KeepAlive` is true changes "real lifetime". 
The behavior is tested in <https://github.com/ppy/osu/blob/b7afea37c4f8facce6a43e362371c2470110572c/osu.Game.Tests/Gameplay/TestSceneDrawableHitObject.cs#L43>.
I could add a comment, though.

Reply:
I think this behavior is confusing and I want it to be dropped. But I'll leave this as is to keep the scope of this PR minimal.